### ISSUE TITLE
Removing the markdown toggle

### ISSUE THREAD

**Issue by philipjohn** [Automattic/simplenote-electron#259]
Once I enabled markdown in settings I expected it to just work in my notes, so I was surprised that an extra step was needed to further enable it on individual notes. That feels like more effort than should be necessary.

**Comment by drw158** [comment]
After turning on the global MD setting, would you expect every new subsequent note created to have MD automatically turned on? Also, what would you expect to happen if you turned off the global MD setting once had created several MD notes?

Thanks for testing!

**Comment by philipjohn** [comment]
I could have been less succinct, sorry :)

I would have at least expected new notes to have Markdown enabled, but actually I didn't test that they were. (I now have, and see that I do have to enable MD on new notes too.)

As for existing notes, I'm not sure. I wouldn't mind markdown being enabled on them personally. I'm unsure whether it would be a problem or poor UX to have it retrospectively enabled.

> Also, what would you expect to happen if you turned off the global MD setting once had created several MD notes?

I think I'd expect to see them all return to plain text. I'm seeing the global setting as a global setting that will effect all notes. I think that's the attitude behind my expectation that I wouldn't have to do a per-note setting as well.

Hope that helps!

**Comment by Sushubh** [comment]
I had a few MD notes created in the web app so I was confused when they did not appear properly in the just released Windows app. Found the toggle and it processed my existing MD notes properly without touching the non-MD ones.

**Comment by rikai** [comment]
I myself would have thought the global toggle would have been enabled by default.

That use case makes sense to me, because it would provide the full functionality out of the box, and people could then disable markdown support on notes they didn't want markdown on, which i can only imagine would be a minority of use cases, or globally if they dont want markdown at all.

As for the global toggle, i'd expect that the toggle would only affect notes made after the toggle changed. This makes sense for it the intented functionality to me because for existing notes... You've already decided what format you wanted those notes to be in. Why would you want a global setting to override the decision you already made? That sounds like a headache waiting to happen.

**Comment by roundhill** [comment]
+1 from me to have new notes have markdown enabled if the global was switched.

In hindsight, I think we probably don't need to worry about users seeing the markdown toggle even when they don't know what markdown is. I wonder if maybe we should remove the global toggle and always show the toggle in note info. Maybe we could add a setting to always set new notes as markdown enabled as well.

**Comment by Sushubh** [comment]
It would be great if there was a toggle ui element to prevent any confusion. just flick it to turn on/off markdown for the post. There should be no need for a global setting!

**Comment by rikai** [comment]
There is always the option of putting the toggle in the button area above a note somewhere with a mouseover saying that it's a markdown toggle. :)

**Comment by drw158** [comment]
> It would be great if there was a toggle ui element to prevent any confusion. just flick it to turn on/off markdown for the post. There should be no need for a global setting!

This currently exists in the note Info panel.

> There is always the option of putting the toggle in the button area above a note somewhere with a mouseover saying that it's a markdown toggle. :)
 
We try to put all settings like that in the note Info panel.

**Comment by drw158** [comment]
After reviewing all of the feedback, I'm inclined to:

- Only have one global MD switch, in account preferences
- Global switch would affect all notes if turned on/off
- Eliminate the MD switch in the note Info panel

Pros
- Simpler, it would clear up the issue reported here

Cons
- Notes could not be individually set for MD

Having a MD setting in the note Info panel will inherently require another setting, to determine if new notes will have the per note setting on/off by default. Creating a setting for another setting is too convoluted and I think the pros outlined above outweigh the cons.

Basically, we'd be hurting users who want only some notes with MD enabled, but we are making the process much more simple and easier to understand.

**Comment by EJOweb** [comment]
I agree with avoiding another setting. A global MD switch would suit me perfectly. I can't picture a situation where I want some notes to particularly have MD turned off.

**Comment by roundhill** [comment]
I like your plan Dave but it is a problem not to have the toggle in order to set the markdown flag, which is a per-note thing in the back end. This way the markdown formatting persists across apps and also renders the published note in markdown if it were flagged as so.

I think with this in mind, we should leave the individual toggle but have all notes created be markdown enabled if the previous note had it enabled. This way we could get rid of the global setting.

**Comment by MiteshNinja** [comment]
On the flip side, I would hate for a single global switch to toggle markdown ON/OFF on every post.

Since the primary scope and functionality of SN is to be able to take notes, there are tons of us who don't use markdown. 

Any personal note I write is written in plain text. I only use markdown for much larger pieces like blog posts, longer more structured class notes etc. Also, enabling the markdown support occupies extra space for the preview/edit options. 

I'd say the best option would be to remove global toggle and keep the individual one; defaulting to markdown disabled. Most of the time markdown is useful only when the note isn't very small, so IMHO it justifies the 1 extra second it takes to manually enable markdown on those notes.

Just my opinion.

**Comment by dmsnell** [comment]
I've been using the new Day One 2 recently and I like the way they have their editor. Like Simplenote, it's just text, but they _always_ have a distinction between editing and viewing. In that context, Markdown is always enabled for them. Although we have historical reasons not to enable markdown everywhere due to previous notes being written without that expectation, I could see a good compromise of the following:

 - No more global markdown toggle
 - Every note keeps the markdown toggle
 - If a note is created with markdown enabled, all further new notes automatically have it enabled
 - New accounts have it enabled by default

**Comment by drw158** [comment]
Thank you @roundhill, @dmsnell, and @MiteshNinja for the feedback. Y'all have made a good case for a per-note MD toggle, in the Info panel. I'll have to check out Day One 2.

Sounds like going forward, we should:

- Remove the global MD toggle.
- Keep the per-note MD toggle.

then there are 3 options:

- Default for toggle is ON.
- Default for toggle is OFF.
- The toggle state is determined by the note you have selected.

I think we are getting closer! All three options for the toggle state are valid. The next time we have some resources to push out a new release, we should take care of this.

**Comment by anothernode** [comment]
I just started using simplenote because I took notice that it supports markdown. Now, at least using the electron based app, I have to enable markdown manually for every new note I create. I would definitely appreciate it very much if there would be a way for me to have markdown enabled by default for new notes I create.
